File: Factory/FactoryMethod/Java/src/Pizza/ChicagoStyle/ChicagoStyleClamPizza.java
package eu.svetlin.design.patterns.factory.factorymethod.Pizza.ChicagoStyle;

import eu.svetlin.design.patterns.factory.factorymethod.Pizza.Pizza;

public class ChicagoStyleClamPizza extends Pizza {
    public ChicagoStyleClamPizza(){
        name = "Chicago Style Clam Pizza";
        dough = "Extra Thick Crust Dough";
        sauce = "Plum Tomato Sauce";

        toppings.add("Shredded Mozzarella Cheese");
        toppings.add("Frozen Clams from Chesepeake Bay");
    }

    @Override
    public void cut() {
        System.out.println("Cutting the pizza into Square slices");
    }
}

File: Factory/FactoryMethod/TypeScript/src/Pizza/ChicagoStyle/ChicagoStyleClamPizza.ts
import Pizza from '../Pizza';

class ChicagoStyleClamPizza extends Pizza {
  constructor() {
    super();
    this.name = 'Chicago Style Clam Pizza';
    this.dough = 'Extra Thick Crust Dough';
    this.sauce = 'Plum Tomato Sauce';

    this.toppings.push('Shredded Mozzarella Cheese');
    this.toppings.push('Frozen Clams from Chesepeake Bay');
  }

  cut(): void {
    console.log('Cutting the pizza into sqaure slices');
  }
}

export default ChicagoStyleClamPizza;
